cordapp/src/main/kotlin/net/corda/plugins/CordappPlugin.kt

```diff
@@ -100,33 +106,79 @@ class CordappPlugin : Plugin<Project> {
         jarTask.dependsOn(task)
     }
 
+    private fun isCordappProject(project: Project): Boolean{
+        return try{
+            project.plugins.getPlugin(net.corda.plugins.CordappPlugin::class.java)
+            true
+        }catch (e: UnknownPluginException){
+            false
+        }
+    }
+
+    private fun configurePomCreation(project: Project) {
+        project.gradle.taskGraph.beforeTask { task ->
```

Comment:
Scary!!! I am sure you _don't_ want to be iterating over Gradle's _entire task graph_ every time you apply the `cordapp` plugin inside a module.

Reply:
because of the way that we apply the maven-publish plugin indirectly via the publish-utils project, the generateMavenPom task is not available when the cordapp plugin is applied, but rather it is only available after gradle has configured itself.

I have not been able to figure out a way to latch onto the generateMavenPom task without using this approach.

Reply:
This is the problem that all Gradle Plugin Writers are perpetually bashing their heads against - you cannot control the order in which Gradle executes `apply` blocks, and configuration information from users isn't available until all plugins have been applied and evaluation is complete.

Reply:
Why doesn't this work?
```kotlin
project.tasks.forEach { task ->
    if (task.name.startsWith("generatePomFileFor")) {
```

Reply:
Perhaps even:
```kotlin
project.tasks.filter { it.name.startsWith("generatePomFile") }.forEach { task ->
```

Reply:
Or possibly:
```kotlin
project.configure(project.tasks) { task ->
    if (task.name.startsWith("generatePomFile")) {
        task.doFirst { aboutToExecute ->
```